django/contrib/admindocs/views.py

```diff
@@ -411,8 +412,8 @@ def simplify_regex(pattern):
     """
     pattern = replace_named_groups(pattern)
     pattern = replace_unnamed_groups(pattern)
-    # clean up any outstanding regex-y characters.
-    pattern = pattern.replace('^', '').replace('$', '').replace('?', '')
+    # Remove any outstanding regex metacharacters.
+    pattern = re.sub(r'(\\[bBAZ]|[?^$+*])', '', pattern)
```

Comment:
Is there a reason you don't need to take into account whether the initial `'\'` is part of an escaped `'\'`, for example in the pattern `r'a\\Z'`?

Reply:
Hey!

`r'a\\Z` is expanded as `'a\\\\Z'`. `re.sub(r'(\\[bBAZ]|[?^$+*])', '', r"a\\Z")` (i.e. `re.sub(r'(\\[bBAZ]|[?^$+*])', '', "a\\\\Z")`) results in `'a\\'`. I think that this is indeed the correct output. Please let me know if I'm missing something and the result should be something else instead.

Reply:
In the pattern I provided, the `'Z'` is a plain `'Z'` (because the backslash preceding it is escaped), but your substitution line is removing it.

Reply:
Good shout. We can use a negative look behind assertion: `(?<!\\)\\[bBAZ]`.

Reply:
Yes. However, note that isn't sufficient to replace all occurrences of e.g. `r'\Z'`, for example `r'a\\\Z'`.